formatcode.sh

```diff
@@ -14,27 +19,42 @@ set -o nounset
 # Get CPU count
 OS=$(uname)
 NPROC=1
-if [[ $OS = "Linux" || $OS = "Darwin" ]] ; then
-    NPROC=$(getconf _NPROCESSORS_ONLN)
+if [[ ${OS} = "Linux" ]] ; then
+    NPROC=$(nproc)
+elif [[ ${OS} = "Darwin" ]] ; then
+    NPROC=$(sysctl -n hw.physicalcpu)
 fi
 
 # Discover clang-format
 if type clang-format-10 2> /dev/null ; then
     CLANG_FORMAT=clang-format-10
-elif type clang-format-8 2> /dev/null ; then
-    CLANG_FORMAT=clang-format-8
-else
+elif type clang-format 2> /dev/null ; then
+    # Clang format found, but need to check version
     CLANG_FORMAT=clang-format
+    V=$(clang-format --version)
+    if [[ $V != "*version 10.0*" ]]; then
+        echo "clang-format is not 10.0 (returned ${V})"
+        #exit 1
```

Comment:
Why it didn't exit here ?

Reply:
Good point, it was commented out in the original source and I left it as-is. But it makes sense to enable it, because clang-format-11+ is in the wild already and not compatible with clang-format-10 rules.